libwacom/libwacom-database.c

```diff
@@ -266,25 +266,38 @@ libwacom_parse_stylus_keyfile(WacomDeviceDatabase *db, const char *path)
 		stylus->eraser_type = eraser_type_from_str (type);
 		g_free (type);
 
-		if (stylus->eraser_type == WACOM_ERASER_NONE) {
-			stylus->has_eraser = g_key_file_get_boolean(keyfile, groups[i], "HasEraser", &error);
-			if (error && error->code == G_KEY_FILE_ERROR_INVALID_VALUE)
-				g_warning ("Stylus %s (%s) %s\n", stylus->name, groups[i], error->message);
-			g_clear_error (&error);
-			stylus->has_lens = g_key_file_get_boolean(keyfile, groups[i], "HasLens", &error);
-			if (error && error->code == G_KEY_FILE_ERROR_INVALID_VALUE)
-				g_warning ("Stylus %s (%s) %s\n", stylus->name, groups[i], error->message);
-			g_clear_error (&error);
-			stylus->has_wheel = g_key_file_get_boolean(keyfile, groups[i], "HasWheel", &error);
-			if (error && error->code == G_KEY_FILE_ERROR_INVALID_VALUE)
-				g_warning ("Stylus %s (%s) %s\n", stylus->name, groups[i], error->message);
-			g_clear_error (&error);
-		} else {
-			stylus->has_eraser = FALSE;
-			stylus->has_lens = FALSE;
-			stylus->has_wheel = FALSE;
+		string_list = g_key_file_get_string_list (keyfile, groups[i], "PairedStylusIds", NULL, NULL);
+		if (string_list) {
+			GArray *array;
+			guint i;
+
+			array = g_array_new (FALSE, FALSE, sizeof(int));
+			for (i = 0; string_list[i]; i++) {
```

Comment:
`i` is already used for the surrounding `groups` iteration. Which is fine because it'll be shadowed but we shouldn't do this anyway, not least because the `g_warning` below will print the wrong `groups[i]`. (This issue is currently in place with the axis type parsing too, see #248)

Reply:
Fixed